modules/apps/object/object-js-components-web/src/main/resources/META-INF/resources/components/Select/SingleSelect.tsx

```diff
@@ -31,35 +32,69 @@ export function SingleSelect<
 >({onChange = () => {}, options, children, ...otherProps}: IProps<T>) {
 	const [dropdownActive, setDropdownActive] = useState<boolean>(false);
 	const arrayChildren = Children.toArray(children);
+	const [showPopover, setShowPopover] = useState(false);
+
+	useEffect(() => {
+		if (!dropdownActive) {
+			setShowPopover(false);
+		}
+	}, [dropdownActive]);
 
 	return (
 		<BaseSelect
 			dropdownActive={dropdownActive}
 			setDropdownActive={setDropdownActive}
 			{...otherProps}
 		>
-			{options.map((option, index) => (
-				<ClayDropDown.Item
-					className={
-						option.type
-							? 'lfr-object__single-select--with-label'
-							: ''
-					}
-					key={index}
-					onClick={() => {
-						setDropdownActive(false);
-						onChange(option);
-					}}
-				>
-					<div>{option.label}</div>
+			{options.map((option, index) => {
+				let events = {};
+				if (option.popover) {
```

Comment:
@alinedoleron We need a blank line between the definition of events and its usage.